lib/handlers/injector-sor.ts

```diff
@@ -137,13 +140,19 @@ export abstract class InjectorSOR<Router, QueryParams> extends Injector<
             break
         }
 
-        const provider = new ethers.providers.JsonRpcProvider(
-          {
-            url: url,
-            timeout,
-          },
-          chainId
-        )
+        const provider: ethers.providers.BaseProvider = new DefaultEVMClient({
+          allProviders: [
+            new InstrumentedEVMProvider({
+              url: {
+                url: url,
+                timeout,
+              },
+              network: chainId,
+              name: ProviderName.INFURA,
```

Comment:
We might need to figure out a way to link this provider name to the url. We don't have any guarantee right now that Infura is the actual provider name for all of our chains, but we are assuming that.

We don't have to do that right now, but that's some food for thought.

Reply:
I asked ChatGPT to translate the Java equivalent of enum with private field into typescript:

```
public enum Planet {
    MERCURY (3.303e+23, 2.4397e6),
    VENUS   (4.869e+24, 6.0518e6),
    EARTH   (5.976e+24, 6.37814e6),
    MARS    (6.421e+23, 3.3972e6),
    JUPITER (1.9e+27,   7.1492e7),
    SATURN  (5.688e+26, 6.0268e7),
    URANUS  (8.686e+25, 2.5559e7),
    NEPTUNE (1.024e+26, 2.4746e7);

    private final double mass;   // in kilograms
    private final double radius; // in meters
    Planet(double mass, double radius) {
        this.mass = mass;
        this.radius = radius;
    }
    private double mass() { return mass; }
    private double radius() { return radius; }

    // universal gravitational constant  (m3 kg-1 s-2)
    public static final double G = 6.67300E-11;

    double surfaceGravity() {
        return G * mass / (radius * radius);
    }
    double surfaceWeight(double otherMass) {
        return otherMass * surfaceGravity();
    }
    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Usage: java Planet <earth_weight>");
            System.exit(-1);
        }
        double earthWeight = Double.parseDouble(args[0]);
        double mass = earthWeight/EARTH.surfaceGravity();
        for (Planet p : Planet.values())
           System.out.printf("Your weight on %s is %f%n",
                             p, p.surfaceWeight(mass));
    }
}
```

and this is what I got:

```
class Planet {
    private mass: number;   // in kilograms
    private radius: number; // in meters

    // Universal gravitational constant (m3 kg-1 s-2)
    private static readonly G: number = 6.67300E-11;

    // Define enum members
    public static readonly MERCURY = new Planet(3.303e+23, 2.4397e6);
    public static readonly VENUS = new Planet(4.869e+24, 6.0518e6);
    public static readonly EARTH = new Planet(5.976e+24, 6.37814e6);
    public static readonly MARS = new Planet(6.421e+23, 3.3972e6);
    public static readonly JUPITER = new Planet(1.9e+27, 7.1492e7);
    public static readonly SATURN = new Planet(5.688e+26, 6.0268e7);
    public static readonly URANUS = new Planet(8.686e+25, 2.5559e7);
    public static readonly NEPTUNE = new Planet(1.024e+26, 2.4746e7);

    // All Planets
    public static readonly ALL_PLANETS: Planet[] = [
        Planet.MERCURY,
        Planet.VENUS,
        Planet.EARTH,
        Planet.MARS,
        Planet.JUPITER,
        Planet.SATURN,
        Planet.URANUS,
        Planet.NEPTUNE,
    ];

    constructor(mass: number, radius: number) {
        this.mass = mass;
        this.radius = radius;
    }

    private getMass(): number { 
        return this.mass; 
    }

    private getRadius(): number { 
        return this.radius; 
    }

    surfaceGravity(): number {
        return Planet.G * this.getMass() / (this.getRadius() * this.getRadius());
    }

    surfaceWeight(otherMass: number): number {
        return otherMass * this.surfaceGravity();
    }
}

// You can't use command line arguments in the browser, so we use a placeholder value
let earthWeight: number = 100;
let mass: number = earthWeight / Planet.EARTH.surfaceGravity();

for (let p of Planet.ALL_PLANETS) {
    console.log(`Your weight on ${p} is ${p.surfaceWeight(mass)}`);
}
```

I think we can probably do it later.